File: cpp-tiny/lib/Models/EmojiReaction.cpp


#include "EmojiReaction.h"

using namespace Tiny;

EmojiReaction::EmojiReaction()
{
	emoji_code = null;
	emoji_name = null;
	reaction_type = null;
	user_id = null;
	user = null;
}

EmojiReaction::EmojiReaction(std::string jsonString)
{
	this->fromJson(jsonString);
}

EmojiReaction::~EmojiReaction()
{

}

void
EmojiReaction::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *emoji_codeKey = "emoji_code";
    
    if(object.has_key(emoji_codeKey)) 
    {
        bourne::json value = object[emoji_codeKey];


        

        AnyType* obj = &emoji_code;
		obj->fromJson(value.dump());

    }

    const char *emoji_nameKey = "emoji_name";
    
    if(object.has_key(emoji_nameKey)) 
    {
        bourne::json value = object[emoji_nameKey];


        

        AnyType* obj = &emoji_name;
		obj->fromJson(value.dump());

    }

    const char *reaction_typeKey = "reaction_type";
    
    if(object.has_key(reaction_typeKey)) 
    {
        bourne::json value = object[reaction_typeKey];


        

        AnyType* obj = &reaction_type;
		obj->fromJson(value.dump());

    }

    const char *user_idKey = "user_id";
    
    if(object.has_key(user_idKey)) 
    {
        bourne::json value = object[user_idKey];


        

        AnyType* obj = &user_id;
		obj->fromJson(value.dump());

    }

    const char *userKey = "user";
    
    if(object.has_key(userKey)) 
    {
        bourne::json value = object[userKey];


        

        AnyType* obj = &user;
		obj->fromJson(value.dump());

    }


}

bourne::json
EmojiReaction::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["emoji_code"] = getEmojiCode().toJson();

    




	object["emoji_name"] = getEmojiName().toJson();

    




	object["reaction_type"] = getReactionType().toJson();

    




	object["user_id"] = getUserId().toJson();

    




	object["user"] = getUser().toJson();


    return object;

}

AnyType
EmojiReaction::getEmojiCode()
{
	return emoji_code;
}

void
EmojiReaction::setEmojiCode(AnyType  emoji_code)
{
	this->emoji_code = emoji_code;
}

AnyType
EmojiReaction::getEmojiName()
{
	return emoji_name;
}

void
EmojiReaction::setEmojiName(AnyType  emoji_name)
{
	this->emoji_name = emoji_name;
}

AnyType
EmojiReaction::getReactionType()
{
	return reaction_type;
}

void
EmojiReaction::setReactionType(AnyType  reaction_type)
{
	this->reaction_type = reaction_type;
}

AnyType
EmojiReaction::getUserId()
{
	return user_id;
}

void
EmojiReaction::setUserId(AnyType  user_id)
{
	this->user_id = user_id;
}

AnyType
EmojiReaction::getUser()
{
	return user;
}

void
EmojiReaction::setUser(AnyType  user)
{
	this->user = user;
}

File: java-vertx-web/src/main/java/org/openapitools/vertxweb/server/model/EmojiReaction.java
package org.openapitools.vertxweb.server.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.openapitools.vertxweb.server.model.EmojiReactionAllOf;
import org.openapitools.vertxweb.server.model.EmojiReactionBase;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class EmojiReaction   {
  
  private Object emojiCode = null;
  private Object emojiName = null;
  private Object reactionType = null;
  private Object userId = null;
  private Object user = null;

  public EmojiReaction () {

  }

  public EmojiReaction (Object emojiCode, Object emojiName, Object reactionType, Object userId, Object user) {
    this.emojiCode = emojiCode;
    this.emojiName = emojiName;
    this.reactionType = reactionType;
    this.userId = userId;
    this.user = user;
  }

    
  @JsonProperty("emoji_code")
  public Object getEmojiCode() {
    return emojiCode;
  }
  public void setEmojiCode(Object emojiCode) {
    this.emojiCode = emojiCode;
  }

    
  @JsonProperty("emoji_name")
  public Object getEmojiName() {
    return emojiName;
  }
  public void setEmojiName(Object emojiName) {
    this.emojiName = emojiName;
  }

    
  @JsonProperty("reaction_type")
  public Object getReactionType() {
    return reactionType;
  }
  public void setReactionType(Object reactionType) {
    this.reactionType = reactionType;
  }

    
  @JsonProperty("user_id")
  public Object getUserId() {
    return userId;
  }
  public void setUserId(Object userId) {
    this.userId = userId;
  }

    
  @JsonProperty("user")
  public Object getUser() {
    return user;
  }
  public void setUser(Object user) {
    this.user = user;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    EmojiReaction emojiReaction = (EmojiReaction) o;
    return Objects.equals(emojiCode, emojiReaction.emojiCode) &&
        Objects.equals(emojiName, emojiReaction.emojiName) &&
        Objects.equals(reactionType, emojiReaction.reactionType) &&
        Objects.equals(userId, emojiReaction.userId) &&
        Objects.equals(user, emojiReaction.user);
  }

  @Override
  public int hashCode() {
    return Objects.hash(emojiCode, emojiName, reactionType, userId, user);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class EmojiReaction {\n");
    
    sb.append("    emojiCode: ").append(toIndentedString(emojiCode)).append("\n");
    sb.append("    emojiName: ").append(toIndentedString(emojiName)).append("\n");
    sb.append("    reactionType: ").append(toIndentedString(reactionType)).append("\n");
    sb.append("    userId: ").append(toIndentedString(userId)).append("\n");
    sb.append("    user: ").append(toIndentedString(user)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
